Bug: Inclusion missing in TF 1.4 BUILD file for mpi component

### System information
- **I'm compiling TF from sources to have AVX512F instruction support**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: **16.04**
- **TensorFlow installed from (source or binary)**: **source (11/14/2017)**
- **TensorFlow version (use command below)**: **1.4.0**
- **Python version**: **3.6.1**
- **Bazel version (if compiling from source)**: **0.7.0**
- **GCC/Compiler version (if compiling from source)**: **5.4.0**
- **CUDA/cuDNN version**: **9.0 /7.0.3**
- **GPU model and memory**: **GTX 1080Ti**
- CPU i9 7900x
- **Exact command to reproduce**:


bazel build --config=mkl  --config=cuda  --copt="-O3" --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=0" --copt="-DOMPI_SKIP_MPICXX"  //tensorflow/tools/pip_package:build_pip_package

### Describe the problem
error in compile/build
....
**ERROR**: /home/tadeusz/temp/TF14a/tensorflow-master/tensorflow/contrib/mpi/BUILD:60:1: undeclared inclusion(s) in rule '//tensorflow/contrib/mpi:mpi_rendezvous_mgr':
this rule is missing dependency declarations for the following files included by 'tensorflow/contrib/mpi/mpi_rendezvous_mgr.cc':
  '/home/tadeusz/temp/TF14a/tensorflow-master/tensorflow/core/distributed_runtime/tensor_coding.h'
In file included from ./tensorflow/core/platform/default/logging.h:24:0,
                 from ./tensorflow/core/platform/logging.h:25,
                 from ./tensorflow/core/lib/gtl/array_slice_internal.h:32,
                 from ./tensorflow/core/lib/gtl/array_slice.h:101,
                 from ./tensorflow/core/lib/strings/str_util.h:23,
                 from ./tensorflow/contrib/mpi/mpi_utils.h:25,
                 from ./tensorflow/contrib/mpi/mpi_rendezvous_mgr.h:33,
                 from tensorflow/contrib/mpi/mpi_rendezvous_mgr.cc:18:
./tensorflow/core/util/tensor_format.h: In instantiation of 'T tensorflow::GetTensorDim(tensorflow::gtl::ArraySlice<T>, tensorflow::TensorFormat, char) [with T = long long int]':
./tensorflow/core/util/tensor_format.h:372:47:   required from here
./tensorflow/core/util/tensor_format.h:340:29: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]

### Source code / logs
the following helps
in file //tensorflow/contrib/mpi/BUILD
```
....
cc_library(
    name = "mpi_rendezvous_mgr",
    srcs = ["mpi_rendezvous_mgr.cc"],
    hdrs = ["mpi_rendezvous_mgr.h"],
    deps = [
        ":mpi_msg_proto_cc",
        ":mpi_utils",
        "//tensorflow/core:core_cpu_internal",
        "//tensorflow/core:framework",
        "//tensorflow/core:gpu_runtime",
        "//tensorflow/core:lib",
        "//tensorflow/core:protos_cc",
        "//tensorflow/core:worker_proto_cc",
        "//tensorflow/core/distributed_runtime:base_rendezvous_mgr",
        "//tensorflow/core/distributed_runtime:session_mgr",
        "//tensorflow/core/distributed_runtime:worker_env",
        "//third_party/mpi",
        "//tensorflow/core/distributed_runtime:tensor_coding",   ## <<<<< add this line <<<<<<<
    ],
)

```

--- a/tensorflow/contrib/mpi/BUILD
+++ b/tensorflow/contrib/mpi/BUILD
@@ -72,6 +72,7 @@ cc_library(
         "//tensorflow/core:worker_proto_cc",
         "//tensorflow/core/distributed_runtime:base_rendezvous_mgr",
         "//tensorflow/core/distributed_runtime:session_mgr",
+        "//tensorflow/core/distributed_runtime:tensor_coding",
         "//tensorflow/core/distributed_runtime:worker_env",
         "//third_party/mpi",
     ],

